src/mindlessgen/molecules/generate_molecule.py

```diff
@@ -371,6 +377,39 @@ def generate_random_coordinates(at: np.ndarray) -> tuple[np.ndarray, np.ndarray]
     return xyz, ati
 
 
+def contract_coordinates(
+    xyz: np.ndarray, ati: np.ndarray, scale_bondlength: float
```

Comment:
`scale_bondlength` can be renamed locally already now. (Maybe `scale_minimal_distance` or similar. We are not necessarily scaling bond lengths in particular!)